lib/blockchain.go

```diff
@@ -4899,12 +4899,22 @@ func (bc *Blockchain) CreateMaxSpend(
 				err,
 				"Blockchain.CreateMaxSpend: Problem getting next nonce: ")
 		}
-
 		feeAmountNanos := uint64(0)
 		prevFeeAmountNanos := uint64(0)
 		for feeAmountNanos == 0 || feeAmountNanos != prevFeeAmountNanos {
 			prevFeeAmountNanos = feeAmountNanos
-			feeAmountNanos = _computeMaxTxV1Fee(txn, minFeeRateNanosPerKB)
+			if bc.BlockTip().Height >= bc.params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight &&
+				!isInterfaceValueNil(mempool) &&
+				mempool.GetFeeEstimator() != nil {
```

Comment:
When would the estimator be nil if the mempool isn't nil? If the mempool is properly initialized, I think it's fine to expect all of its contents to be properly initialized too

Reply:
A couple things here:
1. The mempool passed into this function is actually an interface, so it could be either a DeSoMempool or a PosMempool. PosMempool returns a fee estimator when `GetFeeEstimator` is called, DeSoMempool returns nil.
2. There are some tests that pass in a nil value for the mempool to these functions. This is annoying more than anything else.